File: Desktop/dsa/Search and Sort/insertionSort.cpp
#include <iostream>
using namespace std;

void insertionSort(int* A, const int n){

	for(int i = 0; i < n; i++){

		for(int j = i; j>0; j--){
			if(A[j] > A[j-1]) break;
			int temp = A[j];
			A[j] = A[j-1];
			A[j-1] = temp;
		}
	}
}

int main(){
	
	int A[] = {4, 3, 2, 5, 6, 3, 5};

	insertionSort(A, 7);

	for(int i = 0; i < 7; i++){

		cout<<A[i]<<" ";
	}
	cout<<endl;
	
	return 0;
}

File: Desktop/dsa/Search and Sort/insertionSort.py

#fHandle = open("input.txt", 'r')
#A = [int(num) for num in (fHandle.read()).split()]

def insertionSort(A):

	n = len(A)
	for i in range(n-1):
		
		pos = i+1
		while pos > 0:
			if A[pos] > A[pos-1]:
				break
			
			temp = A[pos]
			A[pos] = A[pos - 1]
			A[pos-1] = temp
			pos -= 1
		
	return A
	
A = [3, 2, 6, 2, 6, 5, 3, 8]
B = insertionSort(A)
print B